8247907: XMLDsig logging does not work

caojoshua opened the pull request:
Not a clean backport, due to the following:
* [warn(String s, Object... o)](https://github.com/openjdk/jdk11u-dev/blob/f700d37d2b3d8c23d50defc19853c68bb60027ab/src/java.xml.crypto/share/classes/com/sun/org/slf4j/internal/Logger.java#L80-L83) was not included in the original patch, and was added to tip in a later patch
* Use HexDumpEncoder instead of HexPrinter, because HexPrinter does not exist in JDK11
* Different copyright years

Changes pass the new jtreg test.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8247907](https://bugs.openjdk.org/browse/JDK-8247907): XMLDsig logging does not work ⚠️ Issue is not open.


### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev pull/1129/head:pull/1129` \
`$ git checkout pull/1129`

Update a local copy of the PR: \
`$ git checkout pull/1129` \
`$ git pull https://git.openjdk.org/jdk11u-dev pull/1129/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1129`

View PR using the GUI difftool: \
`$ git pr show -t 1129`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/1129.diff">https://git.openjdk.org/jdk11u-dev/pull/1129.diff</a>

</details>

phohensee reviewed (changes requested):
The copyright date maintainer policy as I understand it is to update the year to what is in the original commit if that's later than what's in the target file, not to the current year. So in this case, use 2020 rather than 2022.

Please document in this PR the use of "warn(String s, Object... o)" in 11 that doesn't exist in 16.

Otherwise lgtm.

caojoshua commented:
> The copyright date maintainer policy as I understand it is to update the year to what is in the original commit if that's later than what's in the target file, not to the current year. So in this case, use 2020 rather than 2022.
> 
> Please document in this PR the use of "warn(String s, Object... o)" in 11 that doesn't exist in 16.
> 
> Otherwise lgtm.

Thanks for the review. I changed the years to 2020 and updated the original PR description

phohensee reviewed (approved):
Lgtm. Please tag the issue with jdk11u-fix-request and a Fix Request (11u) comment.

phohensee commented:
/sponsor

Failing pre-submit test appears unrelated.
